MoltenVK/MoltenVK/API/mvk_datatypes.h

```diff
@@ -414,13 +414,13 @@ MTLWinding mvkMTLWindingFromSpvExecutionMode(uint32_t spvMode);
 MTLTessellationPartitionMode mvkMTLTessellationPartitionModeFromSpvExecutionMode(uint32_t spvMode);
 
 /**
- * Returns the combination of Metal MTLRenderStage bits corresponding to the specified Vulkan VkPiplineStageFlags,
+ * Returns the combination of Metal MTLRenderStage bits corresponding to the specified Vulkan VkPipelineStageFlags2,
  * taking into consideration whether the barrier is to be placed before or after the specified pipeline stages.
  */
-MTLRenderStages mvkMTLRenderStagesFromVkPipelineStageFlags(VkPipelineStageFlags vkStages, bool placeBarrierBefore);
+MTLRenderStages mvkMTLRenderStagesFromVkPipelineStageFlags(VkPipelineStageFlags2 vkStages, bool placeBarrierBefore);
 
-/** Returns the combination of Metal MTLBarrierScope bits corresponding to the specified Vulkan VkAccessFlags. */
-MTLBarrierScope mvkMTLBarrierScopeFromVkAccessFlags(VkAccessFlags vkAccess);
+/** Returns the combination of Metal MTLBarrierScope bits corresponding to the specified Vulkan VkAccessFlags2. */
+MTLBarrierScope mvkMTLBarrierScopeFromVkAccessFlags(VkAccessFlags2 vkAccess);
```

Comment:
This is a breaking change.

Reply:
> This is a breaking change.

It shouldn't be. The spec includes:

> The VkPipelineStageFlags2 bitmask goes beyond the 31 individual bit flags allowable within a C99 enum, which is how [VkPipelineStageFlagBits](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#VkPipelineStageFlagBits) is defined. The first 31 values are common to both, and are interchangeable.

> The VkAccessFlags2 bitmask goes beyond the 31 individual bit flags allowable within a C99 enum, which is how [VkAccessFlagBits](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#VkAccessFlagBits) is defined. The first 31 values are common to both, and are interchangeable.

So `VkPipelineStageFlagBits ` and `VkAccessFlagBits` should pass transparently, and be consistent with previous behaviour.

Am I missing something?

Reply:
On *x86*, loading a 32-bit register clears the upper 32 bits of the corresponding 64-bit register. Does *ARM* work that way, too?

/me looks

...apparently, it does. Never mind, then.